server/src/main/java/org/elasticsearch/search/lookup/FieldValues.java

```diff
@@ -37,6 +38,17 @@
      * @return the value fetcher
      */
     static ValueFetcher valueFetcher(FieldValues<?> fieldValues, SearchExecutionContext context) {
+        return valueFetcher(fieldValues, v -> v, context);
+    }
+
+    /**
+     * Creates a {@link ValueFetcher} that fetches values from a {@link FieldValues} instance
+     * @param fieldValues the source of the values
+     * @param formatter   a function to format the values
+     * @param context the search execution context
+     * @return the value fetcher
+     */
+    static ValueFetcher valueFetcher(FieldValues<?> fieldValues, Function<Object, Object> formatter, SearchExecutionContext context) {
```

Comment:
the plan is to reuse this for dates too?

Reply:
Exactly